### PR TITLE
Feature/soa weini

### PR DESCRIPTION
**Pull request by kapphire**

### CODE REVIEW

File: scrapers/scrape_statements.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ps = con.findAllNext('p')[1:]
+print(len(ps))
+data = []
+count = 0
+new = {
+    'url': '',
+    'link': '',
+    'link_text': '',
+    'bill_number': '',
+    'date_issued': '',
```
Comment (leedavidr) [inline scrapers/scrape_statements.py:23]: minor- I prefer issue_date but we don't have to change

File: scrapers/scrape_statements.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    'congres': '',
```
Comment (leedavidr) [inline scrapers/scrape_statements.py]: typo congress

File: scrapers/scrape_statements.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+
+for i in range(len(ps)):
+    
+    if ps[i].text.find('Administration Policy on') != -1:
+        
+        new['date_issued'] = ps[i].text.split('on')[-1]
```
Comment (leedavidr) [inline scrapers/scrape_statements.py:32]: minor preference, issue_date
below references apply as well new['date_issued']

File: server_py/flatgov/bills/models.py [hunk 4]
```diff
@@ -155,4 +155,20 @@ class Sponsor(models.Model):
...
-        return self.name
\ No newline at end of file
+        return self.name
+
+
+
+class Statement(models.Model):
+    bill_number = models.CharField(max_length=127)
+    bill = models.CharField(max_length=250)
+    congres = models.CharField(max_length=10)
```
Comment (leedavidr) [inline server_py/flatgov/bills/models.py]: typo congress

File: server_py/flatgov/bills/models.py [hunk 5]
```diff
@@ -155,4 +155,20 @@ class Sponsor(models.Model):
...
     def __str__(self):
-        return self.name
\ No newline at end of file
+        return self.name
+
+
+
+class Statement(models.Model):
+    bill_number = models.CharField(max_length=127)
+    bill = models.CharField(max_length=250)
```
Comment (leedavidr) [inline server_py/flatgov/bills/models.py]: not super clear what's stored here, maybe it's the combined name?

File: server_py/flatgov/bills/models.py [hunk 6]
```diff
@@ -155,4 +155,20 @@ class Sponsor(models.Model):
...
+    congres = models.CharField(max_length=10)
+    date_issued = models.CharField(max_length=35)
+    pdf_link = models.CharField(max_length=255, null=True, blank=True)
+    link = models.CharField(max_length=255, null=True, blank=True)
+
+    created_at = models.DateTimeField(auto_now_add=True)
```
Comment (leedavidr) [inline server_py/flatgov/bills/models.py]: minor preference, something like created_date_time

File: server_py/flatgov/templates/bills/detail.html [hunk 7]
```diff
@@ -306,8 +306,36 @@ <h6 class="pt-3">
...
+        <div class="tab-pane fade" id="statement-admin" role="tabpanel" aria-labelledby="statement-admin-tab" style="max-height: 50vh; overflow: scroll;">
+            <h6 class="pt-3">
+                View all current sponsors of {{ bill.type_abbrev }} {{ bill.meta.number }} in the {{ bill.meta.congress }} Congress
+            </h6>
+            <table class="table mt-3" id="statements-table">
+                <thead>
+                    <tr>
+                        <th scope="col">Bill Number</th>
+                        <th scope="col">Bill</th>
+                        <th scope="col">Congres</th>
```
Comment (leedavidr) [inline server_py/flatgov/templates/bills/detail.html]: typo Congress

File: server_py/flatgov/templates/bills/detail.html [hunk 8]
```diff
@@ -306,8 +306,36 @@ <h6 class="pt-3">
...
+                        <th scope="col">Link</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {% for statement in statements %}
+
+                    <tr>
+                        <td>{{ statement.bill_number|truncatechars:10}}</td>
+                        <td style="width: 300px;">{{ statement.bill|truncatechars:120 }}</td>
+                        <td>{{ statement.congres }}</td>
```
Comment (leedavidr) [inline server_py/flatgov/templates/bills/detail.html]: Typo congress

File: server_py/flatgov/templates/bills/detail.html [hunk 9]
```diff
@@ -306,8 +306,36 @@ <h6 class="pt-3">
...
+                        <th scope="col">Congres</th>
+                        <th scope="col">Date Issued</th>
+                        <th scope="col">Link</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {% for statement in statements %}
+
+                    <tr>
+                        <td>{{ statement.bill_number|truncatechars:10}}</td>
```
Comment (leedavidr) [inline server_py/flatgov/templates/bills/detail.html]: do we need to truncate this? what cases are long? seems like an important identifier

File: server_py/flatgov/templates/bills/detail.html [hunk 10]
```diff
@@ -306,8 +306,36 @@ <h6 class="pt-3">
...
+                        <td style="width: 300px;">{{ statement.bill|truncatechars:120 }}</td>
```
Comment (leedavidr) [inline server_py/flatgov/templates/bills/detail.html]: not sure what this data is but same question, is truncation necessary here

File: server_py/flatgov/templates/bills/detail.html [hunk 11]
```diff
@@ -306,8 +306,36 @@ <h6 class="pt-3">
...
+                        <td>{{ statement.congres }}</td>
+                        <td>{{ statement.date_issued }}</td>
```
Comment (leedavidr) [inline server_py/flatgov/templates/bills/detail.html:333]: minor preference, issue_date

File: README.adoc [hunk 12]
```diff
@@ -253,6 +262,11 @@ Updates to the data are done through the Celery taskrunner (see https://docs.cel
 ```bash
 $ pyenv activate flatgov
 $ cd ~/.../FlatGov/server_py/flatgov 
+./manage.py loaddata statements_datas.json
```
Comment (leedavidr) [inline README.adoc:265]: data is already plural, maybe filename should be renamed only if it's easy

File: server_py/flatgov/bills/models.py [hunk 13]
```diff
@@ -160,3 +160,22 @@ class Sponsor(models.Model):
...
         return self.name
+
+
+
+class Statement(models.Model):
+    bill_number = models.CharField(max_length=127)
+    bill_id = models.CharField(max_length=127, null=True, blank=True)
+    bill_title = models.CharField(max_length=250)
+    congress = models.CharField(max_length=10)
+    date_issued = models.CharField(max_length=35)
```
Comment (leedavidr) [inline server_py/flatgov/bills/models.py:171]: minor preference, issue_date

File: server_py/flatgov/bills/models.py [hunk 14]
```diff
@@ -160,3 +160,22 @@ class Sponsor(models.Model):
...
+    permanent_pdf_link = models.FileField(upload_to='statements/', blank=True, null=True)
+    original_pdf_link = models.CharField(max_length=255, null=True, blank=True)
+
+    date_fetched = models.DateTimeField(auto_now_add=True)
```
Comment (leedavidr) [inline server_py/flatgov/bills/models.py:175]: minor preference, fetch_date

**Review by leedavidr — COMMENTED** [review]
Just some comments, I commented on Ayesha's commits as well
If it's easy I would change them, but if it's hard, comment on it and we can work on it later

But names in DB are hard to change later
same with JSON names and they could become standard across the project

**Comment by aih** [comment]
This was handled in a separate PR that has been merged. Closing.

**Pull request closed without merging**